### PR TITLE
feat: add `x-origin` property

### PR DESCRIPTION
**Pull request by aeworxet**
This PR adds the generation of the property `x-origin` during the dereferencing process.
Property `x-origin` is used for origin tracing in `Bundler` and component naming in `Optimizer`.

It originated from [this comment](https://github.com/asyncapi/bundler/issues/97#issuecomment-1330501758) in a year-long discussion: 

> The $ref usually also carries a semantical meaning to understand easier what it is (example "$ref : financial-system.yaml#/components/schemas/bankAccountIdentifier"). If the bundling just resolves this ref inline, the semantical meaning of the $ref pointer gets lost and cannot be recovered in later steps. The optimizer would need to invent an artificial component name for the "bankAccountIdentifier" when moving it to the components section.

Thus, property `x-origin` contains historical values of dereferenced `$ref`s, which are also used by `Optimizer` starting from its version v1.0.0, to give meaningful names to components it moves through the AsyncAPI Document.

Partial resolution of https://github.com/asyncapi/bundler/issues/141

### CODE REVIEW

**Comment by aeworxet** [comment]
@KhudaDad414 @derberg 
@asyncapi/bounty_team

**Review by KhudaDad414 — COMMENTED** [review]
@aeworxet  Tests are failing for some reason. 🤔

**Comment by savvyintegrations** [comment]
For anyone waiting on this PR who doesn't want to fork, you can use `@apidevtools/json-schema-ref-parser` in your local script before calling `bundler`, i.e., just dereference first, e.g.:
```
import bundle from '@asyncapi/bundler';
import $RefParser from "@apidevtools/json-schema-ref-parser";
import { writeFileSync } from 'fs';
import { resolve } from "node:path"
import { mkdir } from "node:fs/promises"

process.chdir("specs")

const srcFileName = 'events.asyncapi.yml';
const outputBase = resolve("../generated/docs/specs") + "/"
const outputPath = `${outputBase}${srcFileName}`;

try {
    await mkdir(outputBase, {recursive: true})
    const clonedSchema = await $RefParser.dereference(srcFileName, {
        dereference: {
            excludedPathMatcher: (path) => {
                console.log('Excluded path', path);
                return path.includes("operations")
            },
        },
        mutateInputSchema: false
    });
    const document = await bundle([clonedSchema], {
        referenceIntoComponents: true,
    });
    writeFileSync(outputPath, document.yml());
    console.log(`Bundled ${srcFileName} to ${outputPath}`)
} catch (err) {
    console.error(err);
}
```

**Comment by aeworxet** [comment]
This PR is ready for merge, but **it should be merged only AFTER https://github.com/asyncapi/optimizer/pull/216 is merged (it will be merged only to `next` not `master`) and `Optimizer` `1.0.0` is released**, due to the transfer of functionality of moving components into `components` to `Optimizer` starting from its version v1.0.0.

File: src/index.ts [hunk 1]
```diff
@@ -74,33 +72,18 @@ import { resolveV3Document } from './v3/parser';
...
-    options.base = toJS(options.base);
-    await parse(options.base);
-  }
-
   const parsedJsons = files.map(file => toJS(file)) as AsyncAPIObject[];
 
-  if (typeof options.baseDir !== 'undefined') {
-    parsedJsons.forEach(parsedJson => resolveBaseFileDir(parsedJson, options.baseDir));
-  }
-
```
Comment (Souvikns) [inline src/index.ts:87]: You are removing `baseDir` function?

File: src/index.ts [hunk 2]
```diff
@@ -74,33 +72,18 @@ import { resolveV3Document } from './v3/parser';
  *
  */
 export default async function bundle(files: string[], options: any = {}) {
-  if (typeof options.base !== 'undefined') {
-    options.base = toJS(options.base);
-    await parse(options.base);
-  }
-
```
Comment (Souvikns) [inline src/index.ts:81]: Why are removing these options?

**Review by Souvikns — CHANGES_REQUESTED** [review]
I don't think you should remove some of the previous changes.

File: src/index.ts (hunk 2 above)
Comment (aeworxet) [reply src/index.ts:81]: `baseDir` and `resolveBaseFileDir()` were related to resolving `$ref`s in `$.channels.*.messages.*` and `$.operations.*.messages.*` only: full dereferencing functionality was rewritten and is now handled by `@apidevtools/json-schema-ref-parser` and I didn't catch usecases when specifying an absolute path was needed. Have I missed something?

Comment (aeworxet) [reply src/index.ts:81]: Is this related to
https://github.com/asyncapi/bundler/issues/136
?

Comment (Souvikns) [reply src/index.ts:81]: Yeah, I see in this PR, you have removed the changes from #136, which is all alright if you are improving the solution. 


When the spec files were in a separate directory, bundler was having problem resolving the paths. So `baseDir` was to solve that issue to let bundler know where to look for the files.

Comment (aeworxet) [reply src/index.ts]: I returned and improved the `baseDir` functionality, but now it is required to pass to the `Bundler` paths of AsyncAPI Documents as an array of strings. Bundler changes the working directory to `baseDir` **and only then** starts reading the AsyncAPI Documents from the filesystem.
Should this change be mentioned somewhere, taking into account that `Bundler` is only v0.5.0?

Comment (aeworxet) [reply src/index.ts]: Added explanation of how `baseDir` currently works to [README.md](https://github.com/asyncapi/bundler/pull/147/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R218).

**Review by Souvikns — APPROVED** [review]
LGTM 🚀

**Pull request merged**